### PR TITLE
NixOS Fixes

### PR DESCRIPTION
**Pull request by hlandau**
I've been using these for ages. Probably time to get them upstreamed.

Another fix to come for pick-to-branch when using worktrees.

### CODE REVIEW

**Review by levitte — APPROVED** [review]

File: release-tools/stage-release.sh [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+#! /usr/bin/env bash
 # Copyright 2020-2023 The OpenSSL Project Authors. All Rights Reserved.
 #
 # Licensed under the Apache License 2.0 (the "License").  You may not use
 # this file except in compliance with the License.  You can obtain a copy
 # in the file LICENSE in the source distribution or at
 # https://www.openssl.org/source/license.html
 
+set -e
+
```
Comment (levitte) [inline release-tools/stage-release.sh:10]: Side note: the move of `-e` was a bit surprising...  `/usr/bin/env bash -e` is as perfectly valid, and I find it more visible.  But, I'm ok with this, still approving this PR.

Comment (hlandau) [reply release-tools/stage-release.sh:10]: Is it? UNIX shebangs are weird... some UNIX platforms allow an arbitrary number of arguments on the shebang line, but many (including Linux) are weird and make everything a single argument, basically only splitting on the first space. So `#!/bin/foo alpha beta gamma` is equivalent to `/bin/foo "alpha beta gamma"`.

Comment (mspncp) [reply release-tools/stage-release.sh:10]: FWIW: personally, I prefer the more verbose line

```
set -o errexit
```

Comment (mspncp) [reply release-tools/stage-release.sh:10]: Don't know whether that's a bashism though.

Comment (paulidale) [reply release-tools/stage-release.sh:10]: The original and the best: `set -e` please.

Comment (mspncp) [reply release-tools/stage-release.sh:10]: ... says the former participant of the IOCCC 😀

**Comment by t8m** [comment]
Merged. Thank you.

**Pull request closed without merging**